crates/sui-adapter/src/execution_engine.rs

```diff
@@ -328,9 +331,9 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
 
     if result.is_err() {
         tracing::error!(
-            "Failed to execute advance epoch transaction. Switching to safe mode. Error: {:?}. System state object: {:?}. Tx data: {:?}",
+            "Failed to execute advance epoch transaction. Switching to safe mode. Error: {:?}. Input objects: {:?}. Tx data: {:?}",
             result.as_ref().err(),
-            temporary_store.read_object(&SUI_SYSTEM_STATE_OBJECT_ID),
+            temporary_store.objects(),
```

Comment:
Good idea to include the input system state obj here. But I believe `temporary_store.read_object(&SUI_SYSTEM_STATE_OBJECT_ID)` would give you the contents of the obj after writes. Do we want to include that here too?

Reply:
But if we hit safe_mode, then the system state object won't be written, right?
Also, now that you asked, I realized that we unfortunately won't be able to get the actual system state object because it's a dynamic field, so it doesn't show up in the input

Reply:
I guess I could dump them after we succeed with safe_mode execution

Reply:
> I realized that we unfortunately won't be able to get the actual system state object because it's a dynamic field, so it doesn't show up in the input

You are right..Technically the dynamic field id can be computed at this point? Can we get it from the store? 

> I guess I could dump them after we succeed with safe_mode execution

If we do this we wouldn't be able to catch the bug we encountered since the safe_mode function just assigns epoch number without checking

Reply:
Probably not here but before we execute this. I will see what I can do

Reply:
Also safe mode function now calls this:
https://github.com/MystenLabs/sui/blob/13747b0a497c7487bbfd7545c7d242ba72931a68/crates/sui-framework/sources/governance/sui_system.move#L891-L896
So it's less safe now...

Reply:
hmm, if we cannot load the inner system state, we cannot do anything?

Reply:
Printed the object content prior to execution.